File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int j = 0,res = 0,mul = 1;
        if(k <= 1) return 0;
        for(int i = 0; i<nums.size(); i++){
            mul *= nums[i];
            while(mul >= k){
                mul /= nums[j++];
            }
            res += i - j + 1;
        }
        return res;
    }
};

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numSubarrayProductLessThanK = function(nums, k) {
    
    let ans = 0,mul = 1,j = 0;
    let i = 0;
    if(k <= 0) return 0;
    for(; i<nums.length; i++){
        let num = nums[i];
        mul *= num;
        while(j < i && mul >= k){
            mul = Math.floor(mul/nums[j++]);
        }
        if(mul < k)
            ans += i - j + 1;
    }
    return ans;
    
};